src/controller/data_model/BUILD.gn

```diff
@@ -22,4 +22,5 @@ chip_data_model("data_model") {
   zap_pregenerated_dir = "gen"
 
   use_default_client_callbacks = true
+  allow_circular_includes_from = [ "${chip_root}/src/controller" ]
```

Comment:
This is codesmell.

Assuming this is because the way zap code is generated?
@vivien-apple - why does zap require/generate circular includes? This makes layering unclear.

Reply:
Yup - zap includes controller as a public dep.

Reply:
> This is codesmell.
> 
> Assuming this is because the way zap code is generated?
> @vivien-apple - why does zap require/generate circular includes? This makes layering unclear.

This is because ZAP generates `CHIPClusters.*` files. The base class for those is located in `src/controller/CHIPCluster.*`. (Note the `s`)